### PR TITLE
MoE support

### PR DESCRIPTION
**Pull request by awan-10**
Co-authored-by: Ammar Ahmad Awan <ammar.awan@microsoft.com>
Co-authored-by: Reza Yazdani <reyazda@microsoft.com>
Co-authored-by: Conglong Li <conglong.li@gmail.com>
Co-authored-by: Jeff Rasley <jerasley@microsoft.com>
Co-authored-by: yaozhewei <zheweiy@berkeley.edu>
Co-authored-by: Adam Moody <moody20@llnl.gov>
Co-authored-by: Minjia Zhang <minjiaz@microsoft.com>

### CODE REVIEW

File: megatron/model/language_model.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 """Transformer based language model."""
 
+from json import encoder
```
Comment (jeffra) [inline megatron/model/language_model.py:18]: is this used? if not, let's remove it

**Review by jeffra — APPROVED** [review]
Looks good to me, a few minor comments then good to merge :)

File: megatron/model/language_model.py (hunk 1 above)
Comment (conglongli) [reply megatron/model/language_model.py]: fixed

File: megatron/tokenizer/gpt2_tokenization.py [hunk 2]
```diff
@@ -281,7 +281,7 @@ def encode(self, text):
         return self.convert_tokens_to_ids(self.tokenize(text))
 
     def decode(self, tokens):
-        text = ''.join([self.decoder[token] for token in tokens])
+        text = ''.join([self.decoder[token%50000] for token in tokens])
```
[the remark this answers is not in the document]
Comment (conglongli) [reply megatron/tokenizer/gpt2_tokenization.py]: fixed

**Pull request merged**